src/scenario-share/ScenarioShareModal.tsx

```diff
@@ -0,0 +1,199 @@
+import { ascending } from "d3-array";
+import React, { useEffect, useState } from "react";
+import styled from "styled-components";
+
+import {
+  addScenarioUser,
+  getScenarioUsers,
+  removeScenarioUser,
+} from "../database";
+import Colors from "../design-system/Colors";
+import InputButton from "../design-system/InputButton";
+import InputText from "../design-system/InputText";
+import Loading from "../design-system/Loading";
+import Modal from "../design-system/Modal";
+import { Spacer } from "../design-system/Spacer";
+import useCurrentUserId from "../hooks/useCurrentUserId";
+import useReadOnlyMode from "../hooks/useReadOnlyMode";
+import useRejectionToast from "../hooks/useRejectionToast";
+import { ScenarioUsers } from "../page-multi-facility/types";
+import useScenario from "../scenario-context/useScenario";
+import AvatarCluster from "./AvatarCluster";
+import ScenarioShareButton from "./ScenarioShareButton";
+import UserAvatar from "./UserAvatar";
+
+const ShareForm = styled.form`
+  align-items: flex-end;
+  display: flex;
+  flex-direction: row;
+`;
+
+const buttonStyles: React.CSSProperties = {
+  width: "78px",
+  fontSize: "13px",
+  fontWeight: "normal",
+};
+
+const UserListContainer = styled.ul`
+  font-size: 13px;
+  font-weight: normal;
+  letter-spacing: -0.02em;
+`;
+
+const UserListItem = styled.li`
+  align-items: center;
+  display: flex;
+  height: 32px;
+  margin: 8px;
+`;
+
+const UserName = styled.div`
+  margin-left: 16px;
+`;
+
+const RemoveButton = styled.button`
+  color: ${Colors.darkRed};
+  margin-left: auto;
+`;
+
+const UserList: React.FC<{
+  users: ScenarioUsers;
+  removeUser: (id: string) => void;
+}> = ({ removeUser, users }) => {
+  const [scenario] = useScenario();
+  const readOnly = useReadOnlyMode(scenario.data);
```

Comment:
We don't render this entire component if we're in `readOnly` mode so these internal `readOnly` checks aren't really necessary

Reply:
That said, we would want these checks in place if we ever want to give read-only users the ability to see who else a scenario has been shared with so I think we should just leave them in

Reply:
yes it is a belt-and-suspenders situation so that it's not making too many assumptions about what context it is used in